src/ForwardDiffNumber.jl

```diff
@@ -128,6 +131,9 @@ isfinite(n::ForwardDiffNumber) = isfinite(value(n))
 isinf(n::ForwardDiffNumber) = isinf(value(n))
 isreal(n::ForwardDiffNumber) = isconstant(n)
 
+floor{T<:Real}(::Type{T}, n::ForwardDiffNumber) = floor(T, value(n))
+ceil{T<:Real}(::Type{T}, n::ForwardDiffNumber) = ceil(T, value(n))
```

Comment:
(accidentally commented in your fork instead of on the PR, so copy and pasted from there...)

Thanks for this! Might want to change the restriction here to T<:ExternalReal if your goal is to disallow T<:ForwardDiffNumber. ForwardDiffNumber was recently subtyped to Real, while ExternalReal is the alias for the union of all the other Real types defined at module load time.